### DISCUSSION TITLE
Limitation on Adding number of cars (cubes) to the Sionna scene

### DISCUSSION THREAD

**Discussion by Ershadh** [NVlabs/sionna#688] [Q&A]
I have created an environment of Urban Macro Setup with roughly a 1000 buildings in a 1000 m x 1000 m area.
Laid 4 roads on x any y axis which are evenly placed across. 
Number of cars I added 15. The cars are represented in the form of plain single component cubes.
When I tried to add any number above 15, the server hangs.
Note: With the 15 cars and all the other elements in the setup the code is still not heavy and finishes scene creation in less than a minute.
It's only  the number of cars which are making it heavy.
 Can someone tell a workaround for this to include as much as 50 cars in the environment?

**Comment by merlinND** [comment]
Hello @Ershadh,

It's difficult to say without a reproducer (complete scene, including meshes + code). Please also make sure that you are testing with the latest version of Sionna.

> the server hangs

When does the server hang (i.e. what code specifically is it executing when you observe it hanging)?

Do you use a GPU, or run on the CPU?
Have you checked memory usage with 15 vs 16 cars? (although running out of memory should result in a crash, not a hang)

**Comment by Ershadh** [comment]
The Sionna version I'm using is 19. However, the problem is that the scene itself is not rendering in the blender.   

I suppose this query is more about blender and less about Sionna. 

However, to answer your queries, I'm using a GPU. Memory usage while rendering >15 cars are around like 50-60 GB.

Thank you Merlin for the response.

**Reply by merlinND** [reply]
Memory usage of 50-60GB sounds high (depending on how much memory your system has). 

> the server hangs

So did you mean that Blender hangs, or that Sionna hangs?
We cannot offer support for Blender here.
